tests/integration/conftest.py

```diff
@@ -627,6 +661,38 @@ def qtest_requirements(self):
         except (RuntimeError, AssertionError):
             raise
 
+    @property
+    def qtest_parent_test_cycles(self):
+        """A list of qTest swagger_client test cycle models which are the direct ancestor for all associated test runs.
+
+        Returns:
+            [swagger_client.TestCycleResource]
+
+        Raises:
+            AssertionError: Associated test cycles do not exist.
+            RuntimeError: General qTest API failure.
+        """
+
+        test_cycle_ids = []
+
+        for run in self.qtest_test_runs:
+            for link in run.links:
+                if link.rel == 'test-cycle':
+                    try:
+                        test_cycle_ids.append(int(self._HREF_ID_REGEX.search(link.href).group(1)))
+                    except IndexError:
```

Comment:
I think you mean `AttributeError`. If the search does not find a match it will return `None` which will not have the attribute `group`.

Reply:
@johnduarte I did some more testing and discovered we're both right. Depending on whether the string is empty, None or contains non-matching text will raise different errors.

Reply:
👍 👍